sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/non_streaming_order_by_aggregator.py

```diff
@@ -0,0 +1,170 @@
+            try:
+                await targetQueryExContext.peek()
+                # if there are matching results in the target ex range add it to the priority queue
+                self._doc_producers.append(targetQueryExContext)
+
+            except StopAsyncIteration:
+                continue
+
+    def _createTargetPartitionQueryExecutionContext(self, partition_key_target_range):
+
+        rewritten_query = self._partitioned_query_ex_info.get_rewritten_query()
+        if rewritten_query:
+            if isinstance(self._query, dict):
+                # this is a parameterized query, collect all the parameters
+                query = dict(self._query)
+                query["query"] = rewritten_query
+            else:
+                query = rewritten_query
+        else:
+            query = self._query
+
+        return document_producer._DocumentProducer(
+            partition_key_target_range,
+            self._client,
+            self._resource_link,
+            query,
+            self._document_producer_comparator,
+            self._options,
+        )
+
+    async def _get_target_partition_key_range(self):
+
+        query_ranges = self._partitioned_query_ex_info.get_query_ranges()
+        return await self._routing_provider.get_overlapping_ranges(
+            self._resource_link, [routing_range.Range.ParseFromDict(range_as_dict) for range_as_dict in query_ranges]
+        )
+
+    async def _configure_partition_ranges(self):
+        # will be a list of (partition_min, partition_max) tuples
+        targetPartitionRanges = await self._get_target_partition_key_range()
+
+        self._document_producer_comparator = document_producer._NonStreamingOrderByComparator(self._sort_orders)
+
+        targetPartitionQueryExecutionContextList = []
+        for partitionTargetRange in targetPartitionRanges:
+            # create and add the child execution context for the target range
+            targetPartitionQueryExecutionContextList.append(
+                self._createTargetPartitionQueryExecutionContext(partitionTargetRange)
+            )
+
+        self._doc_producers = []
+        for targetQueryExContext in targetPartitionQueryExecutionContextList:
+            try:
+                await targetQueryExContext.peek()
+                self._doc_producers.append(targetQueryExContext)
+            except exceptions.CosmosHttpResponseError as e:
+                if exceptions._partition_range_is_gone(e):
+                    # repairing document producer context on partition split
+                    await self._repair_document_producer()
+                else:
+                    raise
+
+            except StopAsyncIteration:
+                continue
+
+        pq_size = self._partitioned_query_ex_info.get_top() or self._partitioned_query_ex_info.get_limit()
+        self._orderByPQ = FixedSizePriorityQueue(pq_size)
+        for doc_producer in self._doc_producers:
+            while True:
```

Comment:
This does seem like an odd implementation for iterating - though I'm not sure what the subtleties of the `peek` operation are here - but feels like this could just be an `async for` loop?

Reply:
since the peek operation can throw errors I opted for doing it this way